From e9dc627a704b2bdf697087f756aa3682cda236af Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Mon, 12 Dec 2022 15:47:36 +0100
Subject: [PATCH 0169/1141] arm64: dts: st: prevent UART RX DMA channel data
 packing on stm32mp251

To avoid potential bytes lost when stopping UART RX DMA channel, prevent
data packing.
Memory accesses will be penalized. But it is the only way to avoid bytes
lost.

Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: I3edec5448a0a90d3d9983885ff2a223cf4cbed99
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/302859
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 8442fd8481bf..22a866915766 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -298,7 +298,7 @@ usart2: serial@400e0000 {
 				reg = <0x400e0000 0x400>;
 				interrupts = <GIC_SPI 115 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART2>;
-				dmas = <&hpdma 11 0x20 0x12 0x0>,
+				dmas = <&hpdma 11 0x20 0x10012 0x0>,
 				       <&hpdma 12 0x20 0x3021 0x0>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_USART2_ID>;
@@ -310,7 +310,7 @@ usart3: serial@400f0000 {
 				reg = <0x400f0000 0x400>;
 				interrupts = <GIC_SPI 116 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART3>;
-				dmas = <&hpdma 13 0x20 0x12 0x0>,
+				dmas = <&hpdma 13 0x20 0x10012 0x0>,
 				       <&hpdma 14 0x20 0x3021 0x0>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_USART3_ID>;
@@ -322,7 +322,7 @@ uart4: serial@40100000 {
 				reg = <0x40100000 0x400>;
 				interrupts = <GIC_SPI 126 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART4>;
-				dmas = <&hpdma 15 0x20 0x12 0x0>,
+				dmas = <&hpdma 15 0x20 0x10012 0x0>,
 				       <&hpdma 16 0x20 0x3021 0x0>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_UART4_ID>;
@@ -334,7 +334,7 @@ uart5: serial@40110000 {
 				reg = <0x40110000 0x400>;
 				interrupts = <GIC_SPI 127 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART5>;
-				dmas = <&hpdma 17 0x20 0x12 0x0>,
+				dmas = <&hpdma 17 0x20 0x10012 0x0>,
 				       <&hpdma 18 0x20 0x3021 0x0>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_UART5_ID>;
@@ -458,7 +458,7 @@ usart6: serial@40220000 {
 				reg = <0x40220000 0x400>;
 				interrupts = <GIC_SPI 136 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART6>;
-				dmas = <&hpdma 19 0x20 0x12 0x0>,
+				dmas = <&hpdma 19 0x20 0x10012 0x0>,
 				       <&hpdma 20 0x20 0x3021 0x0>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_USART6_ID>;
@@ -515,7 +515,7 @@ uart9: serial@402c0000 {
 				reg = <0x402c0000 0x400>;
 				interrupts = <GIC_SPI 150 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART9>;
-				dmas = <&hpdma 25 0x20 0x12 0x0>,
+				dmas = <&hpdma 25 0x20 0x10012 0x0>,
 				       <&hpdma 26 0x20 0x3021 0x0>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_UART9_ID>;
@@ -527,7 +527,7 @@ usart1: serial@40330000 {
 				reg = <0x40330000 0x400>;
 				interrupts = <GIC_SPI 114 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_USART1>;
-				dmas = <&hpdma 9 0x20 0x12 0x0>,
+				dmas = <&hpdma 9 0x20 0x10012 0x0>,
 				       <&hpdma 10 0x20 0x3021 0x0>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_USART1_ID>;
@@ -569,7 +569,7 @@ uart7: serial@40370000 {
 				reg = <0x40370000 0x400>;
 				interrupts = <GIC_SPI 148 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART7>;
-				dmas = <&hpdma 21 0x20 0x12 0x0>,
+				dmas = <&hpdma 21 0x20 0x10012 0x0>,
 				       <&hpdma 22 0x20 0x3021 0x0>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_UART7_ID>;
@@ -581,7 +581,7 @@ uart8: serial@40380000 {
 				reg = <0x40380000 0x400>;
 				interrupts = <GIC_SPI 149 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_UART8>;
-				dmas = <&hpdma 23 0x20 0x12 0x0>,
+				dmas = <&hpdma 23 0x20 0x10012 0x0>,
 				       <&hpdma 24 0x20 0x3021 0x0>;
 				dma-names = "rx", "tx";
 				feature-domains = <&rifsc STM32MP25_RIFSC_UART8_ID>;
-- 
2.39.2

